imap_data_access/file_validation.py

```diff
@@ -0,0 +1,264 @@
+
+        NOTE: There are no optional parameters anymore. All parameters are required.
+        <mission>: imap
+        <instrument>: idex, swe, swapi, hi-45, ultra-45 and etc.
+        <datalevel> : l1a, l1b, l1, l3a and etc.
+        <descriptor>: descriptor stores information specific to instrument. This is
+            decided by each instrument. For L0, "raw" is used.
+        <startdate>: startdate is the earliest date in the data. Format - YYYYMMDD
+        <enddate>: Some instrument and some data level requires to store date range.
+            If there is no end date, then startdate will be used as enddate as well.
+            Format - YYYYMMDD.
+        <version>: This stores software version and data version. Version format is
+            vxx-xx.
+
+        Parameters
+        ----------
+        filename : str | Path
+            Science data filename or file path.
+        """
+        # TODO: Accomodate path or filename
+        self.filename = filename
+
+        try:
+            split_filename = extract_filename_components(self.filename)
+        except ValueError:
+            raise InvalidScienceFileError(
+                f"Invalid filename. Expected file to match format: "
+                f"{FILENAME_CONVENTION}"
+            )
+
+        self.mission = split_filename["mission"]
+        self.instrument = split_filename["instrument"]
+        self.data_level = split_filename["datalevel"]
+        self.descriptor = split_filename["descriptor"]
+        self.startdate = split_filename["startdate"]
+        self.enddate = split_filename["enddate"]
+        self.version = split_filename["version"]
+        self.extension = split_filename["extension"]
+
+        self.error_message = self.validate_filename()
+        if self.error_message:
+            raise InvalidScienceFileError(f"{self.error_message}")
+
+    def validate_filename(self):
+        """Validate the filename and populate the error message for wrong attributes.
+
+        The error message will be an empty string if the filename is valid. Otherwise,
+        all errors with the filename will be put into the error message.
+
+        Returns
+        -------
+        error_message: str
+            Error message for specific missing attribute, or "" if the file name is
+            valid.
+        """
+        error_message = ""
+
+        if any(
+            attr is None or attr == ""
+            for attr in [
+                self.mission,
+                self.instrument,
+                self.data_level,
+                self.descriptor,
+                self.startdate,
+                self.enddate,
+                self.version,
+                self.extension,
+            ]
+        ):
+            error_message = (
+                f"Invalid filename, missing attribute. Filename "
+                f"convention is {FILENAME_CONVENTION} \n"
+            )
```

Comment:
This is a neat way to do it!

Reply:
It was all Tenzin! I slightly modified the method but it was her idea.